### ISSUE TITLE
Make 'Last Name' field in 'Create Lead' form non-mandatory

### ISSUE THREAD

**Issue by SonalikaG** [fatfreecrm/fat_free_crm#999]
Hi, is there a way to make the 'Last Name' field in Create Lead form non-mandatory? Currently, we cannot submit a Lead without providing data in the 'Last Name' field. Please suggest a way.

![image](https://user-images.githubusercontent.com/72000677/175005350-2108d7c2-8002-47e9-9bb9-b76dd6b97d2a.png)

**Comment by srbaker** [comment]
I'm a fan of this, but I think an even better solution is to just merge First Name Last Name into "Name."  Would such a patch be reasonable?

**Comment by johnbumgardner** [comment]
Yeah I like this idea. Probably scales better to non western cultures too

**Comment by johnbumgardner** [comment]
tagging this with good first issue. @srbaker if youre interested in implementation please be our guesr

**Comment by srbaker** [comment]
I'm going to take a crack at this.  Thanks!

**Comment by johnbumgardner** [comment]
> I'm going to take a crack at this. Thanks!

awesome, feel free to reach out to me if you have questions

**Comment by clevertension** [comment]
@johnbumgardner so i think it is good idea to merge First Name and Last Name to Name, so i can implement it

**Comment by ferrisoxide** [comment]
Is merging first and last names potentially a bit "lossy"? I understand that there's some value in just having a name, but are there cases where being able to identify a lead's first name might be useful?

I'm thinking of the use case where you are constructing an email campaign. Reaching out to someone with the greeting "Hi Elizabeth Blackwell" is more formal than "Hi Elizabeth". If you are going for a more personal approach, using someone's full name might not be ideal.

I wonder if just making `last_name` optional might be a more flexible alternative than combing first and last names. Or perhaps the naming scheme could be configurable. I don't necessarily have a solution here - just thinking things through in my own head.

**Comment by ferrisoxide** [comment]
Something else to consider, if we flatten `first_name` and `last_name` to just `name` we will have to do the same for `Contact`, otherwise we won't be abele to convert `Leads` to `Contacts`.

**Comment by CloCkWeRX** [comment]
I'm open to a pull request adding a "display_name" or similar, which can be new/distinct from the structured named elements.
Over time, this can evolve to be the default.

Be aware of
https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/

**Comment by ferrisoxide** [comment]
The original request from @SonalikaG was fairly simple. Is there a danger we'll end up complicating things in the search for a more comprehensive solution that we don't have a current requirement for?

**Comment by mindsublimes** [comment]
@johnbumgardner I’d like to clarify the implementation for the Create Lead form regarding the last_name field. Here are the options we’re considering:

1. Make last_name optional.
2. Merge first_name and last_name into a single name field.
3. Maintain separate first_name and last_name fields.

Could you please let me know what we need to implement?

**Comment by barnaclebarnes** [comment]
@mindsublimes There are 2 settings:

```
:require_first_names: true
:require_last_names: true
```

These don't work however as the migration forces not null on the table:

```
      t.string :first_name,  limit: 64, null: false, default: ""
      t.string :last_name,   limit: 64, null: false, default: ""
```

You should keep separate first and last name fields as you often need first name or last name separately in different systems. I would update the table to allow null on first name and last name.

You will often not even have a first or last name for a lead, you may ONLY have their email address. 

You can add a "display name" field/method that does some logic on first/last/email address.

**Comment by ferrisoxide** [comment]
@mindsublimes @johnbumgardner 

I'd still be arguing for option 1. It's literally what the user asked for - the only issue I can see is that it would have to be extended to `Contact`.

**Comment by barnaclebarnes** [comment]
@ferrisoxide yes, small correction - `first_name` OR `last_name` OR `email` would be required. 

And yes, it should be applied to `Contact` as well.
